src/main/java/com/projects/quickcart/controller/ProductController.java

```diff
@@ -17,6 +18,23 @@ public ModelAndView productsView() {
 		ModelAndView mView = new ModelAndView("products");
 		mView.addObject("producs", bService.getProducts());
 		return mView;
+
 	}
 
+	@GetMapping("/{productId}")
+	public ModelAndView getProductsView(@PathVariable("productId") int productId) {
+		ModelAndView mView = new ModelAndView("products");
+		mView.addObject("products", bService.getProduct());
+
+		return mView;
+
+	}
+
+	@GetMapping("/{category}")
+	public ModelAndView findProductView(@PathVariable("category") String category) {
+		ModelAndView mView = new ModelAndView("products");
+		mView.addObject("products", bService.findProductByCategory());
```

Comment:
the attribute name is `producs` since there is a typo in name from previous commits, so you should follow the same name